### PR TITLE
Enhancement/6251 top performing keywords

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue:

- #6251 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/KeyMetrics/MetricTileTable.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+						}
+					) }
+				</tr>
+			) );
+	} else if ( !! ZeroState ) {
+		tbody = (
+			<tr className="googlesitekit-table__body-row googlesitekit-table__body-row--no-data">
+				<td
+					className="googlesitekit-table__body-item"
+					colSpan={ columns.length }
```
Comment (techanvil) [inline assets/js/components/KeyMetrics/MetricTileTable.js:79]: This `colSpan` doesn't appear to be needed:

```suggestion
```

File: assets/js/components/KeyMetrics/MetricTileTable.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	let tbody = null;
+
+	if ( rows?.length > 0 ) {
+		tbody = rows
+			.slice( 0, limit || rows.length )
+			.map( ( row, rowIndex ) => (
+				<tr key={ rowIndex }>
+					{ columns.map(
+						(
+							{ Component, field, className: columnClassName },
```
Comment (techanvil) [inline assets/js/components/KeyMetrics/MetricTileTable.js:48]: We could do without this alias, or do you prefer to keep it?

```suggestion
							{ Component, field, className },
```

File: assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js [hunk 3]
```diff
@@ -21,31 +21,120 @@
...
+	const columns = [
+		{
+			field: 'keys.0',
+			Component: ( { fieldValue } ) => {
+				const searchAnalyticsURL = useSelect( ( select ) => {
+					return ! viewOnlyDashboard
+						? select( MODULES_SEARCH_CONSOLE ).getServiceReportURL(
+								{
+									...keywordsDateRangeArgs,
+									query: `!${ fieldValue }`,
```
Comment (techanvil) [inline assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js:87]: I find this `!` prefix a bit obscure, would you mind adding an inline comment to clarify it?

File: assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js [hunk 4]
```diff
@@ -21,31 +21,120 @@
...
+import { numFmt } from '../../../../util';
+import Link from '../../../../components/Link';
+import MetricTileTable from '../../../../components/KeyMetrics/MetricTileTable';
+import useViewOnly from '../../../../hooks/useViewOnly';
+import { whenKeyMetricsWidgetVisible } from '../../../../components/KeyMetrics';
+import { ZeroDataMessage } from '../common';
+
+const { useSelect, useInViewSelect } = Data;
+
+function PopularKeywordsWidget( { Widget } ) {
```
Comment (techanvil) [inline assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js:48]: Just an aside to point out that the IB specifies `AnalyticsTopPerformingKeywordsWidget` as the widget name but it's clearly a bit out of date, not to mention the fact that it's a Search Console widget. Just mentioning for the sake of visibility in case anyone else happens upon this, this is all good.

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @eugene-manuilov, this is generally looking good. I have left a few minor comments, please take a look.

Additionally, the IB does mention adding unit test coverage but there is none, has this been overlooked or do you not feel the coverage would add value?

File: assets/js/components/KeyMetrics/MetricTileTable.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+		</Widget>
+	);
+}
+
+MetricTileTable.propTypes = {
+	Widget: PropTypes.elementType.isRequired,
+	loading: PropTypes.bool,
+	title: PropTypes.string,
+	rows: PropTypes.arrayOf(
+		PropTypes.oneOfType( [ PropTypes.array, PropTypes.object ] )
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/components/KeyMetrics/MetricTileTable.js]: Just updated prop types for the MetricTileTable component.

File: assets/js/components/KeyMetrics/MetricTileTable.js (hunk 1 above)
Comment (eugene-manuilov) [reply assets/js/components/KeyMetrics/MetricTileTable.js]: Yep, good catch. Updated.

File: assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js (hunk 3 above)
Comment (techanvil) [reply assets/js/modules/search-console/components/widgets/PopularKeywordsWidget.js:88]: Thanks, sorry to be a bit pedantic though but the comment doesn't really clear things up all that much as it doesn't explain _how_ the SC console uses the query parameter, simply that it _does_ use it which is quite easy to figure out.

I've taken a look and it's evident that the SC uses the `!` prefix to indicate an exact match for the query, rather than say a substring match, so I'd suggest updating the comment accordingly, something like this:

> The exclamation mark at the beginning of the query specifies that the term should be treated as an exact match on the Search Console search results page.

Screengrab of the SC "New query" UI for reference:

image

File: assets/js/components/KeyMetrics/MetricTileTable.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/components/KeyMetrics/MetricTileTable.js]: Ok fair enough, we can address `ReportTable` separately.

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @eugene-manuilov this is looking good, just one small point about the comment, but also you didn't address this point I made in my [previous review summary](https://github.com/google/site-kit-wp/pull/7150#pullrequestreview-1477512951):

> Additionally, the IB does mention adding unit test coverage but there is none, has this been overlooked or do you not feel the coverage would add value?

**Comment by eugene-manuilov** [comment]
> Thanks @eugene-manuilov this is looking good, just one small point about the comment, but also you didn't address this point I made in my [previous review summary](https://github.com/google/site-kit-wp/pull/7150#pullrequestreview-1477512951):
> 
> > Additionally, the IB does mention adding unit test coverage but there is none, has this been overlooked or do you not feel the coverage would add value?

We had some basic tests for widgets at the beginning, but then they all were deleted at some point as there were not enough value in it. We should probably update IB to not mention it anymore. I'll update it shortly.
